Marlin/Configuration_adv.h

```diff
@@ -1529,9 +1529,10 @@
 //
 #define ARC_SUPPORT               // Disable this feature to save ~3226 bytes
 #if ENABLED(ARC_SUPPORT)
-  #define MM_PER_ARC_SEGMENT  1   // Length of each arc segment
-  #define MIN_ARC_SEGMENTS   24   // Minimum number of segments in a complete circle
-  #define N_ARC_CORRECTION   25   // Number of interpolated segments between corrections
+  #define MM_PER_ARC_SEGMENT    1 // (mm) Length (or minimum length) of each arc segment
+  #define MIN_ARC_SEGMENTS     24 // Minimum number of segments in a complete circle
```

Comment:
There is a logic or linguistic error.
`#define MM_PER_ARC_SEGMENT    1 // (mm) Length (or minimum length) of each arc segment` -> No segment can be smaller than 1 mm!
`#define MIN_ARC_SEGMENTS     24 // Minimum number of segments in a complete circle` -> When the circumference of the circle is smaller than 24 mm print a circle with not smaller than that - because the segments cant be smaller then 1mm.

One of this defines is unlikely to be a minimum. Else arcs with small diameters are impossible.

Reply:
MIN_ARC_SEGMENTS overrides MM_PER_ARC_SEGMENT to make possibly shorter segments.  Personally I don't like MIN_ARC_SEGMENTS because it is trying to enforce smoothness irrespective of radius or feed rate, and it can generate excessive resolution that causes stalls.

Reply:
So `MM_PER_ARC_SEGMENT` isn't a minimum but a maximum segment length.

Reply:
With `MIN_STEPS_PER_SEGMENT` we already have a lower limit for the length of one segment.

Reply:
MM_PER_ARC_SEGMENT, when not using ARC_SEGMENTS_PER_SEC, is the 'default' segment length and it can be shorter if MIN_ARC_SEGMENTS implies more smoothness than MM_PER_ARC_SEGMENT provides.

When using ARC_SEGMENTS_PER_SEC, the 'usual' behavior is for segment length to be determined by the feed rate but for very low feed rates the segments can become extremely short.  If the feed rate is low enough that the segments are shorter than MM_PER_ARC_SEGMENT then it uses MM_PER_ARC_SEGMENT instead, so in that sense it is a minimum on the speed-derived segment length.  But the MIN_ARC_SEGMENTS is still in effect, so a very fast small circle might be a hexagon according to ARC_SEGMENTS_PER_SEC, and MIN_ARC_SEGMENTS of 7 or more would force a smoother curve and the segments would be made shorter (and the segment rate would be higher than ARC_SEGMENTS_PER_SEC).

Not all combinations of these settings make sense, for example a large MM_PER_ARC_SEGMENT and a high rate of segments per second would mean the MM_PER_ARC_SEGMENT would almost always take precedence and the segments per second has no effect.  A large value for MIN_ARC_SEGMENTS would mean that small circles will ignore feed rate or (otherwise) minimum resolution and proceed with possibly extremely fine resolution that has no benefit.

Yes MIN_STEPS_PER_SEGMENT causes small segments to collapse, so there is no real benefit resolution-wise for MM_PER_ARC_SEGMENT to be much less than that.

I think a combination of settings that makes sense is for ARC_SEGMENTS_PER_SEC to be a value that the CPU can keep up with comfortably, and for MM_PER_ARC_SEGMENT to be a small value similar to the length that corresponds to MIN_STEPS_PER_SEGMENT.  And MIN_ARC_SEGMENTS should not be used because ARC_SEGMENTS_PER_SEC will already provide the best smoothness that's reasonable at a given feed rate.